### PR TITLE
Partially automate and document the intended flow for backports

### PR DESCRIPTION
**Pull request by PawelLipski**

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -75,18 +75,32 @@ jobs:
...
             source scripts/utils.sh
-            tag=v$(extract_version_from_current_wd)
-            git tag "$tag"
-            git push https://${GITHUB_TOKEN}@github.com/Virtuslab/git-machete-intellij-plugin.git "$tag"
-      - deploy:
-          name: Publish plugin to Jetbrains Marketplace
-          command: ./gradlew publishPlugin
+            version=$(extract_version_from_current_wd)
+            git checkout -b "backport/$version"
+            ssh-keyscan -t rsa github.com | ssh-keygen -lf -
```
Comment (PawelLipski) [inline .circleci/config.yml:94]: So that the pipeline don't stall on the prompt whether we trust the key

File: .circleci/config.yml [hunk 2]
```diff
@@ -75,18 +75,32 @@ jobs:
...
+            hub pull-request \
```
Comment (PawelLipski) [inline .circleci/config.yml:95]: Authorized with `$GITHUB_TOKEN` as well

File: scripts/enforce-pre-release-version-correct-use [hunk 3]
```diff
@@ -11,16 +11,23 @@ if [[ ${CI_PULL_REQUEST-} ]]; then
...
-  if [[ ( $CIRCLE_BRANCH == master || $CIRCLE_BRANCH == release/* || $CIRCLE_BRANCH == hotfix/* ) && $head_pre_release ]]; then
-    die 'Pre-release version is not allowed on master/release/hotfix branch tip'
-  elif [[ $CIRCLE_BRANCH != master && -z $head_pre_release ]]; then
-    die 'Non-master/release/hotfix branch tip must have a pre-release version (a.b.c-d, not a.b.c)'
+  if [[ $CIRCLE_BRANCH == master || $CIRCLE_BRANCH == release/* || $CIRCLE_BRANCH == hotfix/* || $CIRCLE_BRANCH == backport/* ]]; then
+    [[ -z $head_pre_release ]] || \
+      die 'Pre-release version is not allowed on master/release/hotfix/backport branch tip'
+  else
+    [[ $head_pre_release ]] || \
+      die 'Non-master/release/hotfix/backport branch tip must have a pre-release version (a.b.c-d, not a.b.c)'
```
Comment (mkondratek) [inline scripts/enforce-pre-release-version-correct-use:31]: It's a bit unclear what is `non-ed` here :confused:

File: .circleci/config.yml (hunk 2 above)
Comment (mkondratek) [reply .circleci/config.yml:137]: wow, this is cool :rocket:

File: scripts/enforce-pre-release-version-correct-use (hunk 3 above)
Comment (PawelLipski) [reply scripts/enforce-pre-release-version-correct-use:31]: Done, fixed... but I'm lazy, fixed that on the very end of PR chain (#329) :trollface:

**Pull request merged**